### ISSUE TITLE
[Arm64/Ubuntu] Crossgen segmentation faults during build bootstrap

### ISSUE THREAD

**Issue by sdmaclea** [dotnet/coreclr#17007]
While buidling arm64 bootstrap cli with this combination

```
**** DETECTING VERSIONS IN SEED CLI ****
Framework version: 2.1.0-preview2-26316-06
SDK version:       2.1.300-preview2-008375
FXR version:       2.1.0-preview2-26316-06
**** DETECTING GIT COMMIT HASHES ****
coreclr hash:    9d0e7065b788333c712a31d11fc46fe561887b66
corefx hash:     23165cee77ae4dd7a7645b7927872948bb27a73e
core-setup hash: e8d248c21937da58626ef828f6e55abfa22ca8a1
```

I am seeing segmentation faults during crossgen

```
**** Beginning crossgen for arm64 target  ****
_: line 1: 10247 Segmentation fault      timeout $__crossgentimeout $2/crossgen /MissingDependenciesOK /Platform_Assemblies_Paths $2:$(dirname $1) /in $1 /out $1.ni >> $1.log 2>&1
_: line 1: 11712 Segmentation fault      timeout $__crossgentimeout $2/crossgen /MissingDependenciesOK /Platform_Assemblies_Paths $2:$(dirname $1) /in $1 /out $1.ni >> $1.log 2>&1
_: line 1: 13638 Segmentation fault      timeout $__crossgentimeout $2/crossgen /MissingDependenciesOK /Platform_Assemblies_Paths $2:$(dirname $1) /in $1 /out $1.ni >> $1.log 2>&1
_: line 1: 14088 Segmentation fault      timeout $__crossgentimeout $2/crossgen /MissingDependenciesOK /Platform_Assemblies_Paths $2:$(dirname $1) /in $1 /out $1.ni >> $1.log 2>&1
_: line 1: 13977 Segmentation fault      timeout $__crossgentimeout $2/crossgen /MissingDependenciesOK /Platform_Assemblies_Paths $2:$(dirname $1) /in $1 /out $1.ni >> $1.log 2>&1
_: line 1: 12109 Segmentation fault      timeout $__crossgentimeout $2/crossgen /MissingDependenciesOK /Platform_Assemblies_Paths $2:$(dirname $1) /in $1 /out $1.ni >> $1.log 2>&1
_: line 1: 13673 Segmentation fault      timeout $__crossgentimeout $2/crossgen /MissingDependenciesOK /Platform_Assemblies_Paths $2:$(dirname $1) /in $1 /out $1.ni >> $1.log 2>&1

**** Crossgen failed for the following dlls:

/home/vmjenkins/workspace/Dotnet/build-bootstrap/bootstrap/dotnetcli/sdk/2.1.300-preview2-008375/TestHost/System.Collections.Immutable.dll
/home/vmjenkins/workspace/Dotnet/build-bootstrap/bootstrap/dotnetcli/shared/Microsoft.AspNetCore.All/2.1.0-preview2-30338/Microsoft.AspNetCore.SignalR.Redis.dll
/home/vmjenkins/workspace/Dotnet/build-bootstrap/bootstrap/dotnetcli/sdk/2.1.300-preview2-008375/MSBuild.dll
/home/vmjenkins/workspace/Dotnet/build-bootstrap/bootstrap/dotnetcli/shared/Microsoft.AspNetCore.All/2.1.0-preview2-30338/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.dll
/home/vmjenkins/workspace/Dotnet/build-bootstrap/bootstrap/dotnetcli/shared/Microsoft.AspNetCore.App/2.1.0-preview2-30338/Microsoft.AspNetCore.JsonPatch.dll
/home/vmjenkins/workspace/Dotnet/build-bootstrap/bootstrap/dotnetcli/shared/Microsoft.NETCore.App/2.1.0-preview2-26316-06/System.Collections.Immutable.dll
/home/vmjenkins/workspace/Dotnet/build-bootstrap/bootstrap/dotnetcli/shared/Microsoft.NETCore.App/2.1.0-preview2-26316-06/System.Collections.dll
/home/vmjenkins/workspace/Dotnet/build-bootstrap/bootstrap/dotnetcli/sdk/2.1.300-preview2-008375/Microsoft.CodeAnalysis.CSharp.dll
/home/vmjenkins/workspace/Dotnet/build-bootstrap/bootstrap/dotnetcli/shared/Microsoft.AspNetCore.App/2.1.0-preview2-30338/Microsoft.CodeAnalysis.CSharp.dll
```

**Comment by sdmaclea** [comment]
Debugging...

**Comment by sdmaclea** [comment]
Fixed by #17131
